cpp/src/c_api/random_walks.cpp

```diff
@@ -123,6 +123,14 @@ struct node2vec_functor : public abstract_functor {
         // std::make_unique<sampling_params_t>(2, p_, q_, false));
         std::make_unique<sampling_params_t>(cugraph::sampling_strategy_t::NODE2VEC, p_, q_));
 
+      //
+      // Need to unrenumber the vertices in the resulting paths
+      // FIXME: this is not necessary (wasteful) if the input graph was not
+      // renumbered.
+      //
```

Comment:
We may just delete this comment, input graphs are 'by default' renumbered except for some special cases (e.g. debugging).

Reply:
Agree we should delete the comment.

* `unrenumber_local_int_vertices` is very fast (no comms, just a very efficient call to `thrust::transform`)
* Renumbering must be enabled for MG graphs
* For SG graphs it can be turned off in some circumstances, but transposing the graph will turn it on.  There is no bookkeeping in the C API to keep track of what kind of graph transformations have occurred.  So there's no good way to tell, even if you didn't initially renumber the graph, what the state of this current graph is.

Reply:
It will be removed in the next commit